redisinsight/ui/src/components/base/tooltip/RiTooltip.spec.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react'
+import { fireEvent, screen, act } from '@testing-library/react'
+import { render, waitForRiTooltipVisible } from 'uiSrc/utils/test-utils'
+import { RiTooltip, RiTooltipProps } from './RITooltip'
+
+const TestButton = () => (
+  <button type="button" data-testid="tooltip-trigger">
+    Hover me
+  </button>
+)
+
+const defaultProps: RiTooltipProps = {
+  children: <TestButton />,
+  content: 'Test tooltip content',
+}
+
+describe('RiTooltip', () => {
+  it('should render', () => {
+    expect(render(<RiTooltip {...defaultProps} />)).toBeTruthy()
+  })
+
+  it('should render children', () => {
+    render(<RiTooltip {...defaultProps} />)
+
+    expect(screen.getByTestId('tooltip-trigger')).toBeInTheDocument()
+  })
+
+  it('should render tooltip content on focus', async () => {
+    render(<RiTooltip {...defaultProps} />)
+
+    await act(async () => {
+      fireEvent.focus(screen.getByTestId('tooltip-trigger'))
+    })
+    await waitForRiTooltipVisible()
+
+    expect(screen.getAllByText('Test tooltip content')[0]).toBeInTheDocument()
+  })
+
+  it('should render tooltip with title and content', async () => {
+    render(
+      <RiTooltip {...defaultProps} title="Test Title" content="Test content" />,
+    )
+
+    await act(async () => {
+      fireEvent.focus(screen.getByTestId('tooltip-trigger'))
+    })
+    await waitForRiTooltipVisible()
+
+    expect(screen.getAllByText('Test Title')[0]).toBeInTheDocument()
```

Comment:
Question: Why not the simpler query function `getByText`, in this case, when we know there will be only one text element rendered? It looks cleaner to me, and I wonder whether you know any perks of using the `getAllByText` variant?

```suggestion
    expect(screen.getByText('Test Title')).toBeInTheDocument()
```

Reply:
Good to know, thanks! I might face a similar issue at some point and will remember your note (hopefully) 😀